tests/Kernel/MediaRenderTest.php

```diff
@@ -59,45 +62,133 @@ protected function setUp() {
-    $file->setPermanent();
-    $file->save();
+    // Make image media translatable.
+    $setting = $this->container->get('entity_type.manager')->getStorage('language_content_settings')->create([
+      'langcode' => 'en',
+      'statur' => TRUE,
+      'id' => 'media.document',
+      'target_entity_type_id' => 'media',
+      'target_bundle' => 'document',
+      'default_langcode' => 'site_default',
+      'language_alterable' => 'true',
+    ]);
+    $setting->setThirdPartySetting('content_translation', 'enabled', TRUE);
+    $setting->save();
+
+    // Create english file.
+    $english_file = file_save_data(file_get_contents(drupal_get_path('module', 'oe_media') . '/tests/fixtures/sample.pdf'), 'public://test_en.pdf');
+    $english_file->setPermanent();
+    $english_file->save();
+
+    // Create spanish file.
+    $spanish_file = file_save_data(file_get_contents(drupal_get_path('module', 'oe_media') . '/tests/fixtures/sample.pdf'), 'public://test_es.pdf');
+    $spanish_file->setPermanent();
+    $spanish_file->save();
 
+    // View modes to test.
+    $view_modes = ['default', 'oe_theme_main_content'];
+
+    // Create a document media.
+    /** @var \Drupal\media\MediaInterface $media */
     $media = $this->mediaStorage->create([
       'bundle' => 'document',
-      'name' => 'test document',
+      'name' => 'test document EN',
       'oe_media_file' => [
-        'target_id' => $file->id(),
+        'target_id' => $english_file->id(),
       ],
     ]);
 
     $media->save();
 
-    foreach (['default', 'oe_theme_main_content'] as $view_mode) {
+    // Assert tha the media is rendered properly without translations.
+    foreach ($view_modes as $view_mode) {
       $build = $this->mediaViewBuilder->view($media, $view_mode);
       $crawler = new Crawler($this->renderRoot($build));
 
-      // File wrapper.
-      $file_wrapper = $crawler->filter('.ecl-file');
-      $this->assertCount(1, $file_wrapper);
+      // Assert the rendering of the document.
+      $this->assertMainDocumentRendering($crawler, $media->language());
+      // Assert no translations are rendered.
+      $translation_list = $crawler->filter('.ecl-file__translation-list');
+      $this->assertCount(0, $translation_list);
+    }
 
-      // File row.
-      $file_row = $crawler->filter('.ecl-file .ecl-file__container');
-      $this->assertCount(1, $file_row);
+    // Translate the media to spanish.
+    $media_spanish = $media->addTranslation('es', [
+      'name' => 'test document ES',
+      'oe_media_file' => [
+        'target_id' => $spanish_file->id(),
+      ],
+    ]);
+    $media_spanish->save();
 
-      $file_title = $file_row->filter('.ecl-file__title');
-      $this->assertContains('test document', $file_title->text());
+    // Assert that the media is rendered properly with translations in
+    // all available languages.
```

Comment:
I interpret this sentence as you are checking it in all 24 languages available on the site and I am not sure it is really what you do here. I think you are checking for all available translations or translation languages.